### DISCUSSION TITLE
Skipping crc check

### DISCUSSION THREAD

**Discussion by mihaly-sisak** [nih-at/libzip#470] [Using libzip] [answered]
Hi all! Currently profiling my game, loading save files (reading zip files).
Perf flame graph shows me that ~34% of runtime is inflate_fast(), and ~28% of runtime is crc32_little().
Is there any option to skip crc check while reading (zlib) zip files?
Tried to google the issue and looked at docs, could not find anything.

**Comment by 0-wiz-0** [comment, marked as answer]
No, that's not possible. The crc is used to verify that extracting the data was successful (no data corruption occurred).
According to its homepage, (zlib-ng)[https://github.com/zlib-ng/zlib-ng] is a drop-in replacement for zlib which is optimized for modern systems, so you could try using that instead of zlib and check if that improves your flamegraph output.

**Comment by mihaly-sisak** [comment]
Thank you this is a really great find! 
If someone cames across this post and compiles zlib-ng, make sure to set the CMake option ZLIB_COMPAT=ON. Then it is literally a drop-in replacement.
